internal/controller/storageclient_controller.go

```diff
@@ -77,15 +100,47 @@ type storageClientReconcile struct {
 // SetupWithManager sets up the controller with the Manager.
 func (r *StorageClientReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	generationChangePredicate := predicate.GenerationChangedPredicate{}
-	bldr := ctrl.NewControllerManagedBy(mgr).
+	enqueueStorageClients := handler.EnqueueRequestsFromMapFunc(
+		func(ctx context.Context, _ client.Object) []ctrl.Request {
+			storageClients := &v1alpha1.StorageClientList{}
+			if err := r.Client.List(ctx, storageClients); err != nil {
+				return []ctrl.Request{}
+			}
+			requests := make([]ctrl.Request, len(storageClients.Items))
+			for idx := range storageClients.Items {
+				requests[idx] = ctrl.Request{
+					NamespacedName: client.ObjectKeyFromObject(&storageClients.Items[idx]),
+				}
+			}
+			return requests
+		},
+	)
+	controller, err := ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.StorageClient{}).
 		Owns(&batchv1.CronJob{}).
 		Owns(&quotav1.ClusterResourceQuota{}, builder.WithPredicates(generationChangePredicate)).
 		Owns(&nbv1.NooBaa{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Owns(&corev1.Secret{}).
 		Owns(&csiopv1a1.CephConnection{}, builder.WithPredicates(generationChangePredicate)).
-		Owns(&csiopv1a1.ClientProfileMapping{}, builder.WithPredicates(generationChangePredicate))
-	return bldr.Complete(r)
+		Owns(&csiopv1a1.ClientProfileMapping{}, builder.WithPredicates(generationChangePredicate)).
+		Watches(
+			&extv1.CustomResourceDefinition{},
+			enqueueStorageClients,
+			builder.WithPredicates(
+				predicate.NewPredicateFuncs(func(obj client.Object) bool {
+					return slices.Contains(crdsToWatch, obj.GetName())
```

Comment:
Cant we just use the keys of the map for this?

Reply:
done.